src/extractors/auth.rs

```diff
@@ -0,0 +1,92 @@
+use axum::{
+    async_trait,
+    extract::FromRequestParts,
+    http::{request::Parts, HeaderMap, StatusCode},
+};
+use serde::{Deserialize, Serialize};
+use sqlx::SqlitePool;
+use uuid::Uuid;
+
+#[derive(Deserialize, Serialize, sqlx::Type)]
+#[sqlx(transparent)]
+pub struct AuthToken(Uuid);
+
+impl AuthToken {
+    fn new() -> Self {
+        AuthToken(Uuid::new_v4())
+    }
+
+    fn try_parse(input: &str) -> Result<AuthToken, uuid::Error> {
+        let uuid = Uuid::try_parse(input)?;
+        Ok(AuthToken(uuid))
+    }
+}
+
+// Account id
+pub struct Authentication(pub i64);
+
+#[async_trait]
+impl FromRequestParts<SqlitePool> for Authentication {
+    type Rejection = (StatusCode, String);
+
+    async fn from_request_parts(
+        parts: &mut Parts,
```

Comment:
why mut?

Reply:
Makes it possible to use in request handlers